cpp/include/cuspatial/utility.hpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cusparse.h"
+
+namespace detail {
+
+void HANDLE_CUSPARSE_STATUS(cusparseStatus_t status) {
+  if(status != CUSPARSE_STATUS_SUCCESS) {
+    const char* status_string;
+    switch(status) {
+      case CUSPARSE_STATUS_SUCCESS:
+          status_string = "CUSPARSE_STATUS_SUCCESS";
+          break;
+      case CUSPARSE_STATUS_NOT_INITIALIZED:
+          status_string = "CUSPARSE_STATUS_NOT_INITIALIZED";
+          break;
+      case CUSPARSE_STATUS_ALLOC_FAILED:
+          status_string = "CUSPARSE_STATUS_ALLOC_FAILED";
+          break;
+      case CUSPARSE_STATUS_INVALID_VALUE:
+          status_string = "CUSPARSE_STATUS_INVALID_VALUE";
+          break;
+      case CUSPARSE_STATUS_ARCH_MISMATCH:
+          status_string = "CUSPARSE_STATUS_ARCH_MISMATCH";
+          break;
+      case CUSPARSE_STATUS_EXECUTION_FAILED:
+          status_string = "CUSPARSE_STATUS_EXECUTION_FAILED";
+          break;
+      case CUSPARSE_STATUS_INTERNAL_ERROR:
+          status_string = "CUSPARSE_STATUS_INTERNAL_ERROR";
+          break;
+      case CUSPARSE_STATUS_MATRIX_TYPE_NOT_SUPPORTED:
+          status_string = "CUSPARSE_STATUS_MATRIX_TYPE_NOT_SUPPORTED";
+          break;
+      default:
+          status_string = "UNKNOWN";
+    }
+    printf("Cusparse error status %s\n", status_string);
+    CUDF_EXPECTS(status != CUSPARSE_STATUS_SUCCESS, "Fail");
+  }
+}
+
+static void HandleCudaError( cudaError_t err,
+                         const char *file,
+                         int line ) {
+    if (err != cudaSuccess) {
+        printf( "%s in %s at line %d\n", cudaGetErrorString( err ),
+                file, line );
+        exit( EXIT_FAILURE );
+    }
+}
+#define HANDLE_CUDA_ERROR( err ) (detail::HandleCudaError( err, __FILE__, __LINE__ ))
```

Comment:
Call this CUDA_TRY. Is the cudf one in a public header so you can just call it?

Reply:
Due to going in circles on this related to the addition of `cusparseGetErrorName()` to CUDA 10.1, I'd like to leave this for later. There's plenty of time going forward to iterate on improved cuspatial error handling.